tests/dogtag/pytest-ansible/pki/testlib/common/utils.py

```diff
@@ -0,0 +1,562 @@
+#   to use, modify, copy, or redistribute it subject to the terms
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import logging
+import os
+import random
+import re
+import string
+import subprocess
+import sys
+
+if os.path.isfile('/tmp/test_dir/constants.py'):
+    sys.path.append('/tmp/test_dir')
+    import constants
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout)
+
+basic_pki_cmd = 'pki -d {} -c {} -p {} -P http -n "{}" '.format(constants.NSSDB,
+                                                                constants.CLIENT_DIR_PASSWORD,
+                                                                constants.CA_HTTP_PORT,
+                                                                constants.CA_ADMIN_NICK)
+client_cert_del = 'pki -d {} -c {} client-cert-del '.format(constants.NSSDB,
+                                                            constants.CLIENT_DIR_PASSWORD)
+TOPOLOGY = constants.CA_INSTANCE_NAME.split("-")[1]
+
+if TOPOLOGY != '01':
+    CA_SUBJECT = 'CN=CA Signing Certificate,OU={},O={}'.format(constants.CA_INSTANCE_NAME,
+                                                               constants.CA_SECURITY_DOMAIN_NAME)
+else:
+    CA_SUBJECT = 'CN=CA Signing Certificate,OU={},O={}'.format('pki-tomcat',
+                                                               constants.CA_SECURITY_DOMAIN_NAME)
+
+
+def get_random_string(len=10):
+    random_string = ''.join(random.choice(string.ascii_uppercase +
+                                          string.digits +
+                                          string.ascii_letters)
+                            for _ in range(len))
+    return random_string
+
+
+def system_cmd(cmd):
+    """
+    Invoke a shell command on localhost.
+    system_cmd('ls')
+    :returns: A tuple of output, err message, and return code
+    """
+    ret = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                           stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+    out, err = ret.communicate()
+    return out, err, ret.returncode
+
+
+class pki_key_library(object):
+    def __init__(self, **kwargs):
+        self.nssdb = kwargs.get('nssdb', constants.NSSDB)
+        self.db_pass = kwargs.get('db_pass', constants.CLIENT_DIR_PASSWORD)
+        self.host = kwargs.get('host', constants.MASTER_HOSTNAME)
+        self.protocol = kwargs.get('protocol', 'http')
```

Comment:
Why are we forcing `http`? Since PKI 10.6+ I believe we are using `https` as default